test/backend/CMakeLists.txt

```diff
@@ -33,9 +33,16 @@ execute_process(
   ERROR_QUIET
 )
 if (${PYTEST_XDIST_FOUND} EQUAL 0)
-  message(STATUS "Parallel backend tests  : ON")
-  set(BACKEND_TEST_COMMAND "pytest")
-  set(BACKEND_TEST_ARGS "--forked" "-n" "$\{NPROC\}")
+  if (NOT DEFINED NPROC)
+    if (DEFINED ENV{NPROC})
+      set(NPROC $ENV{NPROC} CACHE STRING "Number of processors for pytest" FORCE)
+    else()
+      set(NPROC "auto" CACHE STRING "Number of processors for pytest" FORCE)
```

Comment:
I assume "auto" is basically the number of CPUs? I typically avoid that due to the same reason mentioned in https://github.com/onnx/onnx-mlir/issues/669#issuecomment-839293610. So I would just use a conservative constant like 4.